public/API/API_GetGameExtended.php

```diff
@@ -50,28 +50,97 @@
  *    string    Expiration                date the claim will expire
  */
 
+use App\Community\Models\AchievementSetClaim;
 use App\Platform\Enums\AchievementFlag;
+use App\Platform\Models\Achievement;
+use App\Platform\Models\Game;
+use Carbon\Carbon;
 
-$gameID = (int) request()->query('i');
+$gameId = (int) request()->query('i');
 $flag = (int) request()->query('f', (string) AchievementFlag::OfficialCore);
-getGameMetadata($gameID, null, $achData, $gameData, flag: $flag, metrics: true);
 
-if ($gameData === null) {
+if ($game = Game::with('system')->find($gameId)) {
+    $gameSetClaims = AchievementSetClaim::where('GameID', $gameId)->get();
+}
+
+if (!$game) {
     return response()->json();
+} else {
+    $gameData = [
+        'ID' => $game->ID,
+        'Title' => $game->Title,
+        'ConsoleID' => $game->ConsoleID,
+        'ForumTopicID' => $game->ForumTopicID,
+        'Flags' => null, // Always '0', this is different in the extended endpoint test for some reason
+        'ImageIcon' => $game->ImageIcon,
+        'ImageTitle' => $game->ImageTitle,
+        'ImageIngame' => $game->ImageIngame,
+        'ImageBoxArt' => $game->ImageBoxArt,
+        'Publisher' => $game->Publisher,
+        'Developer' => $game->Developer,
+        'Genre' => $game->Genre,
+        'Released' => $game->Released,
+        'IsFinal' => $game->IsFinal,
+        'RichPresencePatch' => md5($game->RichPresencePatch),
+        'GuideURL' => $game->GuideURL,
+        'Updated' => $game->Updated,
+    ];
 }
 
-if (empty($achData)) {
-    $gameData['Achievements'] = new ArrayObject(); // issue #484 - force serialization to {}
+/* Achievements Map
+    The maps here are return the data in the order that the v1 api expects, originally
+    the endpoint used function calls that made queries with the data in the
+    order that the endpoint returns but the Eloquent collections order the
+    data by how the columns are laid out in the database. This adds a few ms to the
+    response time, though not enough that it has any real impact.
+*/
+
+if (!$game->achievements->where('Flags', $flag)->isEmpty()) {
+    $gameAchievements = $game->achievements->where('Flags', $flag)->keyBy('ID')->map(function ($am) {
```

Comment:
I like this functional approach, but I've discovered this currently suffers from an N+1 query problem. This is largely because of `Achievement::find`, which is happening in the loop.

There is a big optimization opportunity here, which can be found by lifting the achievement queries out of the loop and grabbing all the needed achievements with a single query, perhaps with `::findMany()`.

Reply:
The query now takes place outside of the map and passes a collection to the map to be keyed and returned in the correct order.